### PR TITLE
Randomized pressure and temp

### PR DESCRIPTION
**Pull request by magwo**

### CODE REVIEW

**Comment by magwo** [comment]
Randomization of pressure and temperature.

A future improvement would be to base the randomization on theaters and their seasonal properties. Temperature should probably also be affected by time of day. However, I think it's best to keep this PR simple with a pretty simple randomization, so that we can see it working before adding more sophisticated randomization.

File: game/weather.py [hunk 1]
```diff
@@ -168,12 +170,13 @@ def generate(
         time_of_day: TimeOfDay,
         settings: Settings,
     ) -> Conditions:
+        _start_time=cls.generate_start_time(theater, day, time_of_day, settings.night_disabled)
```
Comment (magwo) [inline game/weather.py]: I moved this to a local variable to prepare for a more sophisticated time-of-day (respecting night_disabled setting) based temperature generation, for a future PR.

**Comment by magwo** [comment]
Screenshot 2021-07-14 at 14 43 07

Screenshot 2021-07-14 at 14 44 52

**Comment by magwo** [comment]
Here is the pydcs code showing that these properties in the mission.weather object are called qnh and season_temperature:
https://github.com/pydcs/dcs/blob/master/dcs/weather.py#L82

**Comment by HILOK** [comment]
thanks @magwo ! glad that someone was able to get on that topic. 

btw, i noticed lately that the DCS ME would only allow a certain range of temperature values, especially in the lower range. on some maps the limit seems to be set rather high, like 15degC. dont know, if it would cause issues, when temperatures are set below the DCS threshold by liberation...

File: game/weather.py [hunk 2]
```diff
@@ -158,6 +158,8 @@ def generate_wind(self) -> WindConditions:
 class Conditions:
     time_of_day: TimeOfDay
     start_time: datetime.datetime
+    pressure: float
+    temperature: float
```
Comment (DanAlbert) [inline game/weather.py]:
```suggestion

    #: Pressure at sea level in inches of mercury.
    pressure: float
    
    #: Temperature at sea level in Celcius.
    temperature: float

```

Should these be part of `Weather` so you can have different ranges based on rain/storm/etc?

File: game/weather.py [hunk 3]
```diff
@@ -210,3 +213,28 @@ def generate_weather(cls) -> Weather:
             list(chances.keys()), weights=list(chances.values())
         )[0]
         return weather_type()
+
+    @classmethod
+    def generate_pressure(cls) -> float:
+        # "Safe" constants based roughly on ME and viper altimeter
```
Comment (DanAlbert) [inline game/weather.py]:
```suggestion
        # "Safe" constants based roughly on ME and viper altimeter.
        # Units are inches of mercury.
```

File: game/weather.py [hunk 4]
```diff
@@ -210,3 +213,28 @@ def generate_weather(cls) -> Weather:
...
+        # to get a more realistic average pressure.
+        average_pressure = 29.92
+        # Use normalvariate to get normal distribution, more realistic than uniform
+        pressure = random.normalvariate(average_pressure, 0.2)
+        return max(SAFE_MIN, min(SAFE_MAX, pressure))
+        
+
+    @classmethod
+    def generate_temperature(cls) -> float:
+        # "Safe" constants based roughly on ME
```
Comment (DanAlbert) [inline game/weather.py]:
```suggestion
        # "Safe" constants based roughly on ME.
        # Temperatures are in Celcius.
```

File: game/weather.py (hunk 2 above)
Comment (magwo) [reply game/weather.py]: Yes, good idea. Will change.

**Review by DanAlbert — APPROVED** [review]

**Review by DanAlbert — APPROVED** [review]

**Pull request merged**